### PR TITLE
Added angular watcher

### PR DESCRIPTION
**Pull request by SAGV**
Then our FE app doesn't need to have any code inside.
However, It still needs that fix:
https://github.com/goabout/goabout-frontend/pull/287

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -32,12 +32,9 @@ More information about the [`#!` notation in URL's](https://support.google.com/w
 
 ### 2. AngularJS Module
 
+That app will automatically inject code to call phantom when angular has finished all his async requests.
```
Comment (mhogerheijde) [inline readme.md:35]: [Nitpick / non-block] 
- That app => This module
- phantom => PhantomJS

File: readme.md [hunk 2]
```diff
@@ -110,7 +107,7 @@ prevent never-ending loops.
 		proxy_set_header X-Download-From  https://goabout.com/
 		proxy_set_header Authorization    Bearer some-predifined-key
 		if ($prerender) {
-		    # Only proyx if $prerender is not set to 0.
+				# Only proyx if $prerender is not set to 0.
```
Comment (mhogerheijde) [inline readme.md:110]: That seems like 1 tab too many? [non-blocking]

File: readme.md [hunk 3]
```diff
@@ -156,25 +153,25 @@ set the proxy_pass rule to that machine.
 
 * key [not set by default]
 
-  A secret. If set any system that wants to access this prerender setup,
-  _must_ provide an Authorization: Bearer <key> header. Obviously this
-  is only safe if used over http*s*.
+	A secret. If set any system that wants to access this prerender setup,
```
Comment (mhogerheijde) [inline readme.md:156]: I liked that the paragraphs were indented at the same level as the first word... :'(

File: server.js [hunk 4]
```diff
@@ -44,12 +45,8 @@ var renderHtml = function(url, cb) {
...
 
     page.onInitialized = function() {
-       page.evaluate(function() {
-            setTimeout(function() {
-                console.debug("setTimeout() rang, calling callPhantom()");
-                window.callPhantom();
-            }, 10000); //If no callPhantom script was called on the page, then wait 10secs and load page anyway.
-        });
+       watch.angular(page); //Remove that if you're not using angular
+       watch.timeout(page);
```
Comment (mhogerheijde) [inline server.js:49]: Cool, this feels like a better approach indeed.

Comment (mhogerheijde) [inline server.js:49]: Should this be "whichever comes first"?

File: watch.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+									 ' this may mean it is not inside ng-app.');
+							}
+							angular.element(el).injector().get('$browser').
+								notifyWhenNoOutstandingRequests(function() {
+									window.callPhantom();
+									console.debug("Angular is Ready! Calling phantom!");
+								});
+						}
+					} catch (err) {
+					 throw new Error('waitForAngular failed!');
```
Comment (mhogerheijde) [inline watch.js:28]: 1. Log the original error?
2. Indentation seems off.

File: watch.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+							});
+						} else {
+							if (!angular.element(el).injector()) {
+								throw new Error('root element body) has no injector.' +
+									 ' this may mean it is not inside ng-app.');
+							}
+							angular.element(el).injector().get('$browser').
+								notifyWhenNoOutstandingRequests(function() {
+									window.callPhantom();
+									console.debug("Angular is Ready! Calling phantom!");
```
Comment (mhogerheijde) [inline watch.js:24]: Feels like these 2 lines should be swapped?

File: server.js [hunk 7]
```diff
@@ -44,12 +45,8 @@ var renderHtml = function(url, cb) {
     }; //For service purposes only
 
     page.onInitialized = function() {
-       page.evaluate(function() {
-            setTimeout(function() {
-                console.debug("setTimeout() rang, calling callPhantom()");
-                window.callPhantom();
-            }, 10000); //If no callPhantom script was called on the page, then wait 10secs and load page anyway.
-        });
+       watch.angular(page); //Remove that if you're not using angular
```
Comment (mhogerheijde) [inline server.js:48]: So if you don't use Angular, your page only gets rendered after 10 seconds?

**Comment by mhogerheijde** [comment]
@SAGV All the comments about the readme are not important. I do have some little questions about the code though.

File: server.js (hunk 7 above)
Comment (SAGV) [inline server.js:48]: then you can add your own watcher or change the delay

File: server.js (hunk 4 above)
Comment (SAGV) [inline server.js:49]: it is so

File: server.js (hunk 7 above)
Comment (mhogerheijde) [inline server.js:48]: Sure. That makes sense. We're not building an any-and-all-should-work thing here. We _know_ that _we_ are using Angular,.. so this is a good solution.

File: server.js (hunk 4 above)
Comment (mhogerheijde) [inline server.js:49]: At first glance I thought the timeout thing, watches if there are any timeouts running on the page, but it isn't.

**Pull request merged**